src/Ssh.php

```diff
@@ -163,7 +172,8 @@ public function getExecuteCommand($command): string
 
         $commandString = implode(PHP_EOL, $commands);
 
-        if (in_array($this->host, ['local', 'localhost', '127.0.0.1'])) {
+        // Unless local SSH connections are allowed, execute command locally
+        if (!$this->allowLocalConnection && in_array($this->host, ['local', 'localhost', '127.0.0.1'])) {
```

Comment:
Could you move this check to a new function called `localConnectionsAllowed` that returns a boolean?

In that function do not use `$$` but make the code as simple as possible by using early returns.

Reply:
Alright! But just to make sure I got it: you'd like to move the 2 checks inside this `if` to a dedicated method ? If so, wouldn't it be `doRunCommandLocally(): bool` instead ? Thanks

Reply:
Yup